src/Digdir.Domain.Dialogporten.Application/Common/Numbers/OrganizationIdentifier.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Digdir.Domain.Dialogporten.Application.Common.Numbers;
+
+public static partial class OrganizationIdentifier
+{
+    public const string NorwegianOrganizationIdentifierPrefix = "urn:altinn:organization:identifier-no::";
+    private static readonly int[] NorwegianOrgNumberWeights = [3, 2, 7, 6, 5, 4, 3, 2];
+
+    [GeneratedRegex(@"urn:altinn:([\w-]{5,20}):([\w-]{4,20})::([\w-]{5,36})", RegexOptions.None, matchTimeoutMilliseconds: 100)]
+    private static partial Regex IdRegex();
+
+    public static bool IsValid(string identifier)
+    {
+        var match = IdRegex().Match(identifier);
+
+        var namespacePart = match.Groups[1].Value;
+        var type = match.Groups[2].Value;
+        var value = match.Groups[3].Value;
+
+        return namespacePart switch
+        {
+            "organization" => ValidateOrganization(type, value),
```

Comment:
Dette burde vært abstrahert til å fungere for alle typer identifikatorer på en måte som følger open-closed principle slik at vi med tiden kan extende til å fungere med org/person identifikatorer med andre nationaliteter og regler. 

lurer på om vi skulle hatt et interface som representerte en identifikator, og implementasjonene av det interfacet var de forskjellige id'ene vi støttet (value objects). Implementasjonene hadde hatt en TryParse funksjon som hadde gått bra dersom id'en som kom inn hadde prefixet den spesifikke id'en ville ha, og selve iden fulgte reglene til den type id. På toppen av det kunne vi hatt et "factory" (kanskje statisk? 🤔) som hadde gitt ut id med korrekt type basert på stringen som kommer inn. Dette factoryet hadde gjort en assembly scanning ved startup etter alle typene som implementerte interfacet, og lagd en tryParseMethodByPrefix dictionary slik at den viste nøyaktig hvilken implementasjon å sende iden til. Kanskje vi skulle droodla på en slik løsning sammen? Jeg ser for meg at dersom vi fortsetter i dette sporet hvor vi ikke konsoliderer konseptet om disse typer IDer, vil det fort kunne bli søl.

Reply:
I like it, takk for veldig gode innspill! La oss ta en titt på det sammen